exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +74,2842 @@ func TestStart(t *testing.T) {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	var orderSubmission = &order.Submit{
+		Exchange: g.Name,
+		Pair: currency.Pair{
+			Delimiter: currency.UnderscoreDelimiter,
+			Base:      currency.BTC,
+			Quote:     currency.USDT,
+		},
+		Side:      order.Buy,
+		Type:      order.Limit,
+		Price:     1,
+		Amount:    1,
+		AssetType: asset.CrossMargin,
+	}
+	_, err := g.SubmitOrder(context.Background(), orderSubmission)
+	if err != nil {
+		t.Errorf("Order failed to be placed: %v", err)
+	}
+}
+
+func TestCancelExchangeOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	var orderCancellation = &order.Cancel{
+		OrderID:       "1",
+		WalletAddress: core.BitcoinDonationAddress,
+		AccountID:     "1",
+		Pair:          currency.NewPair(currency.LTC, currency.BTC),
+		AssetType:     asset.Spot,
+	}
+	err := g.CancelOrder(context.Background(), orderCancellation)
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
+	}
+}
+
+func TestCancelBatchOrders(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	currencyPair := currency.NewPair(currency.LTC, currency.BTC)
+	_, err := g.CancelBatchOrders(context.Background(), []order.Cancel{
+		{
+			OrderID:       "1",
+			WalletAddress: core.BitcoinDonationAddress,
+			AccountID:     "1",
+			Pair:          currencyPair,
+			AssetType:     asset.Spot,
+		}, {
+			OrderID:       "2",
+			WalletAddress: core.BitcoinDonationAddress,
+			AccountID:     "1",
+			Pair:          currencyPair,
+			AssetType:     asset.Spot,
+		}})
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
 	}
 }
 
 func TestGetDepositAddress(t *testing.T) {
-	if areTestAPIKeysSet() {
-		_, err := g.GetDepositAddress(context.Background(), currency.USDT, "", "TRX")
-		if err != nil {
-			t.Error("Test Fail - GetDepositAddress error", err)
-		}
-	} else {
-		_, err := g.GetDepositAddress(context.Background(), currency.ETC, "", "")
-		if err == nil {
-			t.Error("Test Fail - GetDepositAddress error cannot be nil")
-		}
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	_, err := g.GetDepositAddress(context.Background(), currency.USDT, "", "TRX")
+	if err != nil {
+		t.Error("Test Fail - GetDepositAddress error", err)
+	}
+}
+
+func TestGetActiveOrders(t *testing.T) {
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Pairs:     []currency.Pair{currency.NewPair(currency.USDT, currency.BTC)},
+		Type:      order.AnyType,
+		Side:      order.AnySide,
+		AssetType: asset.Spot,
+	}
+	_, err := g.GetActiveOrders(context.Background(), &getOrdersRequest)
```

Comment:
In the return type for order.FilteredOrders. `Order type` is not being set. `ClientOrderID` is not being set from field `text`.

This comes from PRIOR nits [here](https://github.com/thrasher-corp/gocryptotrader/pull/1058/files#r1051690336)  and [here](https://github.com/thrasher-corp/gocryptotrader/pull/1058/files#r1057974833)